Development/nmos/control_protocol_methods.cpp

```diff
@@ -0,0 +1,568 @@
+                ss << U("property: ") << property_id.serialize() << U(" is outside the available range to do GetSequenceItem");
+                return make_control_protocol_error_response(handle, { nc_method_status::index_out_of_bounds }, ss.str());
+            }
+
+            // unknown property
+            utility::stringstream_t ss;
+            ss << U("unknown property: ") << property_id.serialize() << U(" to do GetSequenceItem");
+            return make_control_protocol_error_response(handle, { nc_method_status::property_not_implemented }, ss.str());
+        }
+
+        // Set sequence item
+        web::json::value set_sequence_item(nmos::resources& resources, nmos::resources::iterator resource, int32_t handle, const web::json::value& arguments, get_control_protocol_class_handler get_control_protocol_class, get_control_protocol_datatype_handler, slog::base_gate& gate)
+        {
+            // note, model mutex is already locked by the outter function, so access to control_protocol_resources is OK...
+
+            const auto& property_id = nmos::fields::nc::id(arguments);
+            const auto& index = nmos::fields::nc::index(arguments);
+            const auto& val = nmos::fields::nc::value(arguments);
+
+            slog::log<slog::severities::more_info>(gate, SLOG_FLF) << "Set sequence item: " << property_id.serialize() << " index: " << index << " value: " << val.serialize();
+
+            // find the relevant nc_property_descriptor
+            const auto& property = find_property(parse_nc_property_id(property_id), parse_nc_class_id(nmos::fields::nc::class_id(resource->data)), get_control_protocol_class);
+            if (!property.is_null())
+            {
+                auto& data = resource->data.at(nmos::fields::nc::name(property));
+
+                if (!nmos::fields::nc::is_sequence(property) || data.is_null() || !data.is_array())
+                {
+                    // property is not a sequence
+                    utility::stringstream_t ss;
+                    ss << U("property: ") << property_id.serialize() << U(" is not a sequence to do SetSequenceItem");
+                    return make_control_protocol_error_response(handle, { nc_method_status::invalid_request }, ss.str());
+                }
+
+                if (data.as_array().size() > (size_t)index)
+                {
+                    const auto notification = make_control_protocol_notification(nmos::fields::nc::oid(resource->data), nc_object_property_changed_event_id, { parse_nc_property_id(property_id), nc_property_change_type::type::sequence_item_changed, val, nc_id(index) });
+                    const auto notification_event = make_control_protocol_notification(web::json::value_of({ notification }));
+
+                    modify_resource(resources, resource->id, [&](nmos::resource& resource)
+                    {
+                        resource.data[nmos::fields::nc::name(property)][index] = val;
```

Comment:
Same question as for line 68: if `val` gets std:move'd can this segfault?

Reply:
Thank you @maweit for spotting it, I will correct them all.